unitTests/ui/ResultListTest.ts

```diff
@@ -308,6 +308,31 @@ export function ResultListTest() {
         .then(() => {
           expect(spyResult).toHaveBeenCalledTimes(6);
           expect(spyResults).toHaveBeenCalledTimes(1);
+          expect(spyResults).toHaveBeenCalledWith(
+            jasmine.any(CustomEvent),
+            jasmine.objectContaining({
+              infiniteScrolling: false
+            })
+          );
+          done();
+        });
+    });
+
+    it('should trigger results displayed event with infiniteScrolling set to true if enableInfiniteScroll is true', done => {
+      const data = FakeResults.createFakeResults(6);
+      const spyResults = jasmine.createSpy('spyResults');
+      test.cmp.options.enableInfiniteScroll = true;
+      $$(test.cmp.element).on(ResultListEvents.newResultsDisplayed, spyResults);
+      test.cmp
+        .buildResults(data)
+        .then(elem => test.cmp.renderResults(elem))
+        .then(() => {
+          expect(spyResults).toHaveBeenCalledWith(
+            jasmine.any(CustomEvent),
+            jasmine.objectContaining({
+              infiniteScrolling: true
+            })
+          );
```

Comment:
It would be easier to read if it was using `async/await`

Reply:
Sure